arkouda/numeric.py

```diff
@@ -1722,6 +1725,89 @@ def where(
     return create_pdarray(type_cast(str, repMsg))
 
 
+def matching_shapes(pda_a: pdarray, pda_b: pdarray):
+    """
+    For now, this will only check that two pdarrays have matching sizes.
+    It's included so that we can put placeholders in for the check as
+    we implement n-dimensional array computation.
+
+    Parameters:
+        pda_a : pdarray
+        pda_b : pdarray
+
+    Returns:
+        True if pda_a.size == pda_b.size, else False
+
+    Examples
+    --------
+    >>> a = np.random.randint(0,10,10).astype(np.float64)
+    >>> b = np.random.randint(0,3,10)
+    >>> c = ak.array(a)
+    >>> d = ak.array(b)
+    >>> ak.matching_shapes(c,d)
+    True
+    >>> b = np.random.randint(0,10,5000)
+    >>> d = ak.array(b)
+    >>> ak.matching_shapes(c,d)
+    False
+
+    """
+
+    return pda_a.size == pda_b.size
+
+
+def array_equal(pda_a: pdarray, pda_b: pdarray, equal_nan: bool = False):
+    """
+    Compares two pdarrays for equality.
+    If neither array has any nan elements, then if all elements are pairwise equal,
+    it returns True.
+    If equal_Nan is False, then any nan element in either array gives a False return.
+    If equal_Nan is True, then pairwise-corresponding nans are considered equal.
+
+    Parameters:
+        pda_a : pdarray
+        pda_b : pdarray
+        equal_nan : boolean to determine how to handle nans, default False
+
+    Returns:
+        With string data: False if one array is of type npstr and the other isn't,
```

Comment:
Typically we let the reviewer resolve the conversation so they can check what was changed.